docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,352 @@
+the HTTPS protocol from client connections. All authentication and authorization
+technologies supported by SEP require configuring TLS as the foundation.
+
+Transport Layer Security (TLS) is the successor to its now-deprecated
+predecessor, Secure Sockets Layer (SSL). This page uses the term TLS to refer
+to TLS and SSL.
+
+Approaches
+----------
+
+To configure Trino with TLS support, there are two alternative paths to
+consider:
+
+* Use the :ref:`load balancer or proxy <https-load-balancer>` at your site
+  or cloud environment to terminate HTTPS/TLS. This approach is the simplest and
+  strongly preferred solution.
+
+* Secure the :ref:`Trino server directly <https-trino>`. This requires you
+  to obtain a valid certificate, and add it to the Trino coordinator's
+  configuration.
+
+.. _https-load-balancer:
+
+Use a load balancer to terminate HTTPS/TLS
+------------------------------------------
+
+Your site or cloud environment may already have a load balancer or proxy server
+configured and running with a valid TLS certificate. In this case, you can work
+with your network administration group to set up your Trino server behind the
+load balancer. The load balancer or proxy server accepts TLS connections and
+forwards them to the Trino coordinator, which runs with default HTTP
+configuration on the default port, 8080.
+
+In this case, the Trino cluster is contained in a virtual private network. The
+load balancer adds request headers with the original client information:
+
+.. code-block:: text
+
+  X-Forwarded-Proto: https
+  X-Forwarded-Host: trino.example.com
+  Forwarded:  for=10.0.16.42;proto=https;by=73.159.60.147
+
+To enable processing of forwarded headers, :ref:`the config properties file
+<config_properties>` must include the following:
+
+.. code-block:: text
+
+  http-server.process-forwarded=true
+
+This completes any necessary configuration for using HTTPS with a load balancer.
+Client tools can access Trino with the URL exposed by the load balancer.
+
+.. _https-trino:
+
+Secure Trino directly
+----------------------
+
+Instead of the preferred mechanism of :ref:`using an external load balancer
+<https-load-balancer>`, you can secure the Trino coordinator itself. This
+requires you to obtain and install a TLS certificate and configure Trino to use
+it for any client connections.
+
+Add a TLS certificate
+^^^^^^^^^^^^^^^^^^^^^
+
+Obtain a TLS certificate file (a cert) from one of the following sources:
+
+* Your site's network administration group
+* Your cloud environment provider
+* A domain name registrar, such as Verisign or GoDaddy
+* A certificate vendor
+* A free certificate generator, such as `letsencrypt.org
+  <https://letsencrypt.org/>`_ or `sslforfree.com
+  <https://www.sslforfree.com/>`_
+* A certificate generating command such as ``openssl`` or ``keytool``
+
+Trino supports certificate files in PEM (PKCS #8 and PKCS #12) or JKS format.
```

Comment:
Maybe we need to have PEM and JKS explained or a link to some info

Reply:
Hmm. An explanation of sorts for PEM format immediately follows in the parenthesized references to the two PKCS formats, both of which are linked to Wikipedia pages. I'll try to do something similar for JKS.

Reply:
My bad. The Wikipedia links are on the TLS page. On this page, I'm adding internal references to that TLS page explanation.